invenio_app/config.py

```diff
@@ -89,6 +89,9 @@
 RATELIMIT_GUEST_USER = '1000 per hour;60 per minute'
 """Rate limit for non logged in users."""
 
+APP_THEME = ['bootstrap3']
```

Comment:
For backward compatibility, I think we need to leave bootstrap templates in the place where they are.

```
invenio_accounts/login_user.html
invenio_accounts/semanticui/login_user.html
```

instead of 

```
invenio_accounts/bootstrap3/login_user.html
invenio_accounts/semanticui/login_user.html
```

Reply:
Actually, the directory structure will be something like (the prefix is applied at the top level):

```
templates/bootstrap3/invenio_accounts/login_user.html
templates/semantic-ui/invenio_accounts/semanticui/login_user.html

...or if we don't move the bootstrap templates:

templates/invenio_accounts/login_user.html
templates/semantic-ui/invenio_accounts/semanticui/login_user.html
```

In any of the cases, since we're overriding the templating mechanism at the very core level (i.e. `app.jinja_env.loader`), old `render_template`/`include`/`extends` usage would still work. Someone would have to be doing something weird with templates, like reading a template using `pkg_resources` or similar. Actually even if we didn't introduce the `bootstrap3` folder there's always the fallback to [looking-up the unprefixed](https://github.com/inveniosoftware/invenio-app/pull/66/files#diff-303566f44fdd7aa9200088a343eadc41R81) template name. I guess we can take a vote on this later.

If this goes in Invenio v3.4, we'll be anyways doing a major (minor for us) bump in the modules and their dependencies.

Reply:
What I meant is that if a user has their private template like this:

```jinja
{# mytemplate.html #}
{% extends "invenio_accounts/login_uer.html" %}
```

It should still work in Invenio v3.4, so there's no forced upgrade in order to get on v3.4.

Possible solutions:

1. Leave Bootstrap3 in place.
```
templates/invenio_accounts/login_user.html
templates/semantic-ui/invenio_accounts/semanticui/login_user.html
```

2. Hack the template loader by always include Bootstrap3 as a default):
```
templates/boostrap3/invenio_accounts/login_user.html
templates/semantic-ui/invenio_accounts/semanticui/login_user.html
```

I guess equivalent of something like:

``APP_THEME.append('bootstrap3')``

3. Add an extra template
```
templates/invenio_accounts/login_user.html
templates/boostrap3/invenio_accounts/login_user.html
templates/semantic-ui/invenio_accounts/semanticui/login_user.html
```

with ``templates/invenio_accounts/login_user.html`` containg:

```jinja
{% extends "bootstrap3/invenio_accounts/login_user.html" %]
```

Reply:
Edit: Not sure if solution 3 actually would work :-)

Reply:
> What I meant is that if a user has their private template like this:
> 
> ```
> {# mytemplate.html #}
> {% extends "invenio_accounts/login_uer.html" %}
> ```
> 
> It should still work in Invenio v3.4, so there's no forced upgrade in order to get on v3.4.

If in v3.4 by default we have `APP_THEME = ['bootstrap3']` (in case we go with the `bootstrap3` folders) or otherwise `APP_THEME = None`, it would still work.

What is an issue is if instead of `mytemplate.html` it's called `invenio_accounts/login_user.html`, in which case we get an infinite loop... One solution would be to exempt the "current" template name in the current request from future lookups.